intro-to-oop-contd/projects-with-many-files.md

```diff
@@ -147,40 +171,176 @@ We must _import_ any resources that we need from another module before we can us
+
+```python
+driver = Driver()
+```
+
+Shorter and clearer!
+
+```
+└── ride_share_app
+    ├── __init__.py
+    ├── driver.py
+    ├── passenger.py
+    └── payment
+        ├── __init__.py
+        ├── cash.py
+        └── credit.py
+```
+
+If our package had a folder called `payment` under `ride_share_app` (with an `__init__.py` file), and `payment` had the modules `cash.py` and `credit.py`, then from either of these modules we could access the `Driver` identifier in `driver.py` as:
+
+```python
+from ..driver import Driver
+```
+
+We added another `.` at the beginning, which tells Python to look up one folder. In principle, we can preface the name we give to `from` with as many `.`s as needed to move us up through folders in the package.
+
+### Importing With `from` Works for All Import Methods
+
+Using `from` to name a module, and then importing specific symbols is very useful. It's so useful that we can also use the `from` style to import from built-in modules and modules using their full package name as well!
+
+```python
+from random import randint
+from ride_share_app.driver import Driver
+
+my_random_int = randint(0, 5)
+driver = Driver()
+```
+
+### Handling Duplicate Names
+
+There is one more optional keyword that can be used with `import` statements. What if there are two modules we want to use: `module_one`, and `module_two`, and each of these modules provides an identifier `UsefulClass`? The following doesn't cause an error, but something's not right:
+
+```python
+from module_one import UsefulClass
+from module_two import UsefulClass
+```
 
-Let's consider this file structure, and these two modules:
+After the second `import`, `UsefulClass` only refers to the version from `module_two`. Of course we could import the entire modules and use dot notation to access the appropriate `UsefulClass` as needed, but we can also import the two versions as:
+
+```python
+from module_one import UsefulClass as UsefulClassOne
+from module_two import UsefulClass as UsefulClassTwo
+```
+
+Instead of importing the identifier `UsefulClass` into the current scope and then overwriting it, this second version imports each of the `UsefulClass` variables into two separate identifiers: `UsefulClassOne`, and `UsefulClassTwo`, which we can use as:
+
+```python
+one = UsefulClassOne()
+one.do_something()
+two = UsefulClassTwo()
+two.do_something_else()
+```
+
+### So Many Ways to Import
+
+As we have shown, there are _many_ different syntaxes and strategies we can use to import modules. If a particular import syntax works for a project, it's valid!
+
+Different situations will need to use different ways of importing. So we should be ready to see many styles of importing. We may even encounter styles that we haven't discussed that we might need to look up. That's ok! But we should feel confident that we can recognize importing when we see it!
+
+## Restructuring a Project
```

Comment:
Soft suggestion/question? I found myself craving that this section was framed as an example walkthrough. Your first sentence in this section pretty much states that though. Would it be valuable to rename this to something like:

```suggestion
## Example: Restructuring a Project
```

```suggestion
## Example Walkthrough: Restructuring a Project
```

or something to that effect?